### ISSUE TITLE
graphsage unsupervised 

### ISSUE THREAD

**Issue by lmxhappy** [dmlc/dgl#1580]
need label? if no labels, it is impossible to train graphsage model?

**Comment by mufeili** [comment]
See the [unsupervised training section](https://github.com/dmlc/dgl/tree/master/examples/pytorch/graphsage) in our GraphSAGE example.

**Comment by BarclayII** [comment]
An unsupervised model tries to find patterns from a dataset with little human supervision (i.e. labels).  In the context of learning on graphs, a case could be learning representation of nodes without clear supervision on the class of nodes.

GNNs can be trained to produce node representations without node labels by utilizing the topology of the graph itself as a supervision signal under certain assumptions.  One common assumption is that nodes that are close to each other should have similar representations.  This naturally leads to a ranking-loss-like function that encourage the representations of two adjacent nodes to be similar under some metric (e.g. dot product), while discouraging nodes that are far away to have dissimilar representations.

The unsupervised loss function in the GraphSAGE paper is one way to do that.  The quality of the learned representations is evaluated by training a multiclass logistic regression on those representations using a very small number of labels.  For more details, you can refer to https://github.com/dmlc/dgl/blob/master/examples/pytorch/graphsage/train_sampling_unsupervised.py.
